.github/workflows/custom-script.yml

```diff
@@ -0,0 +1,59 @@
+# Commit a temporary `ci-script.sh` on the desired local branch before running workflow
+name: Run custom script
+
+on:
+  workflow_dispatch:
+    inputs:
+      runner:
+        description: 'GitHub Actions runner'
+        required: true
+        default: 'ubuntu-latest'
+        type: choice
+        options:
+          - ubuntu-latest
+          - buildjet-32vcpu-ubuntu-2204
+          - gpu-bench
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  cpu-script:
+    name: Run custom `ci-script.sh`
+    if: github.event.inputs.runner != 'gpu-bench'
+    runs-on: ${{ github.event.inputs.runner }}
+    steps:
+      - uses: actions/checkout@v4
+      # Install dependencies
+      - uses: actions-rs/toolchain@v1
+      - uses: Swatinem/rust-cache@v2
+      - name: Run input script
+        run: |
+          chmod +x ci-script.sh
+          ./ci-script.sh
+        working-directory: ${{ github.workspace }}
+
+  gpu-script:
+    name: Run custom `ci-script.sh` on GPU
+    if: github.event.inputs.runner == 'gpu-bench'
+    runs-on: [self-hosted, "${{ github.event.inputs.runner }}"]
```

Comment:
Note that we can restrict this to an L4 or similar by specifying the `"NVIDIA L4"` runner tag. It would need a bit more syntax testing